src/concurrencyUtil.cpp

```diff
@@ -0,0 +1,8 @@
+#include "concurrencyUtil.h"
+
+void initLock(pthread_mutex_t* lock) {
+    pthread_mutexattr_t attr;
+    pthread_mutexattr_init(&attr);
+    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
+    pthread_mutex_init(lock, &attr);
+}
```

Comment:
Global function again.
Why not encapsulate `pthread_mutex_t` and related operations in a Mutex class, and place it together with MutexLocker in `mutex.h/cpp`?

Reply:
My bad. I personally prefer top level function (being a Kotlin fan). Will fix.
The question is: should I just put two classes `Mutex` (with one static method) and `MutexLocker` in one `mutex.h/cpp` ?